### DISCUSSION TITLE
How do I properly filter an array?

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2506] [Q&A] [answered]
I know this code is wrong, but here's what I'm trying to achieve:

```js
export async function getFavorites() {
  const id = "oh8q35rn3xvpeky";

  const user = await pb.collection("users").getFirstListItem(`id="${id}"`);
  console.log(user.favorites)
  const favorites = await pb.collection("groupbuys").getFullList({
    filter: `id = "${user.favorites}"`,
  })

  return favorites;
}
```

I get a user, get an array of strings, containing the IDs of favorite items, and then make a second request to filter the results. However, filtering like this doesn't work. This method works for a single item, what's the correct way of doing it for an array?

**Comment by ganigeorgiev** [comment, marked as answer]
You can either use directly [expand](https://pocketbase.io/docs/expanding-relations/) for the `users` collection (if "favorites" is a `relation` field) OR generate a filter with `||` joined conditions:

```js
const user = await pb.collection("users").getOne(id, {
    expand: 'favorites'
});
console.log(user.expand.favorites);

// OR manually query the related favorites records

const user = await pb.collection("users").getOne(id);
const favorites = await pb.collection("groupbuys").getFullList({
    // aka. `id="id1" || id="id2" || ...`
    filter: user.favorites.map((id) => `id="${id}"`).join("||"),
})
```

**Reply by g1sbi** [reply]
The manual query worked fine. I'd like to use expand, however it returns an empty object. (favorites is a relation)

**Reply by ganigeorgiev** [reply]
> I'd like to use expand, however it returns an empty object.

Make sure that the client request satisfies the View API rule of the "favorites" collection.

**Reply by g1sbi** [reply]
that was it. Thanks for the quick replies as always!
